### PR TITLE
Use token count for speed calc

### LINKED ISSUE #33: Fix speed measurements in case of complex Unicode characters
See https://github.com/petals-infra/chat.petals.dev/pull/31#pullrequestreview-1549659585.

### PR DESCRIPTION
**Pull request by Webifi**
Added `token_count` of delta for each respose, since if max_new_tokens > 1 steps would be be more difficult to use for speed calculation.

Fixes #33.

### CODE REVIEW

**Comment by Webifi** [comment]
(Something went very strange when I tried to pull the merge from your remote branch, hence all the repeated changes.)

**Comment by borzunov** [comment]
@Webifi Great job! I'm gonna test it just in case and merge it soon.

File: static/chat.js [hunk 1]
```diff
@@ -181,7 +182,8 @@ function receiveReplica(inputs) {
 
     if (lastMessageTime != null) {
       totalElapsed += performance.now() - lastMessageTime;
-      nRequests++;
+      tokenCount += response.token_count;
+      nRequests ++;
```
Comment (borzunov) [inline static/chat.js]:
```suggestion
      nRequests++;
```

File: websocket_api.py [hunk 2]
```diff
@@ -48,8 +48,10 @@ def ws_api_generate(ws):
 
                 all_outputs = ''
                 delta_q = []
+                steps=0
```
Comment (borzunov) [inline websocket_api.py]:
```suggestion
                steps = 0
```

File: websocket_api.py [hunk 3]
```diff
@@ -78,9 +80,11 @@ def ws_api_generate(ws):
                         logger.info(f"ws.generate.append_retry(), all_outputs={repr(combined)}")
                     else:
                         all_outputs = combined
+                        token_count = len(delta_q + delta)
                         delta_q = []
                         logger.info(f"ws.generate.step(), all_outputs={repr(all_outputs)}, stop={stop}")
-                        ws.send(json.dumps({"ok": True, "outputs": outputs, "stop": stop}))
+                        ws.send(json.dumps({"ok": True, "outputs": outputs, "stop": stop, "steps": steps, "token_count": token_count}))
+                        steps=0
```
Comment (borzunov) [inline websocket_api.py]:
```suggestion
                        steps = 0
```

**Comment by Webifi** [comment]
@borzunov Anything terribly wrong with this PR?

**Comment by borzunov** [comment]
@Webifi Sorry for the slow response, I'm on vacation right now :)

Are you okay with removing `steps` from the response? I feel like this field is a bit confusing, since people may think that it's the overall number of steps from the beginning of the session, etc.

Other than that, I think it's good to go!

**Comment by Webifi** [comment]
Yeah, I agree. I questioned if I should have even put steps in.

I'm having issues pushing changes to my fork again.  I'll try to get this sorted out when I can find some time.

**Comment by Webifi** [comment]
My chat.petals.dev colab isn't working any longer for some reason, even without these changes applied, so I can't fully test the latest change I just pushed until I'm back in the office in a few weeks.

**Comment by borzunov** [comment]
@Webifi Seems to work, merging it. Thanks for your contribution!

**Pull request merged**